receiver/prometheusreceiver/internal/otlp_transaction.go

```diff
@@ -89,6 +93,12 @@ func (t *transaction) Append(ref storage.SeriesRef, labels labels.Labels, atMs i
 		}
 	}
 
+	// For the `target_info` metric we need to convert it to resource attributes.
+	metricName := labels.Get(model.MetricNameLabel)
+	if metricName == targetMetricName {
```

Comment:
should we also check the metric type?

Reply:
We don't get type information passed down to the `Appender` unfortunately :(